Gems/Terrain/Code/Source/TerrainRenderer/Passes/TerrainClipmapComputePass.cpp

```diff
@@ -55,6 +181,13 @@ namespace Terrain
         TerrainFeatureProcessor* terrainFeatureProcessor = scene->GetFeatureProcessor<TerrainFeatureProcessor>();
         if (terrainFeatureProcessor)
         {
+            const TerrainClipmapManager& clipmapManager = terrainFeatureProcessor->GetClipmapManager();
+
+            clipmapManager.GetMacroDispatchThreadNum(
+                m_dispatchItem.m_arguments.m_direct.m_totalNumberOfThreadsX,
+                m_dispatchItem.m_arguments.m_direct.m_totalNumberOfThreadsY,
+                m_dispatchItem.m_arguments.m_direct.m_totalNumberOfThreadsZ);
```

Comment:
This can also be moved to an IsEnabled() override so you can get the number of threads and return false if you don't need to run the pass.